src/nativestore/NodeManager.cpp

```diff
@@ -100,19 +143,45 @@ RelationBlock *NodeManager::addRelation(NodeBlock source, NodeBlock destination)
     RelationBlock *newRelation = NULL;
     if (source.edgeRef == 0 || destination.edgeRef == 0 ||
         !source.searchRelation(destination)) {  // certainly a new relation block needed
-        newRelation = RelationBlock::add(source, destination);
+        RelationBlock *relationBlock = new RelationBlock(source,destination);
+        newRelation = relationBlock->add(source, destination);
         if (newRelation) {
-            source.updateRelation(newRelation);
-            destination.updateRelation(newRelation);
+            source.updateRelation(newRelation, false);
+            destination.updateRelation(newRelation,false);
+        } else {
+            node_manager_logger.error("Error while adding the new edge/relation for source = " +
+                                      std::string(source.id) + " destination = " + std::string(destination.id));
+        }
+    }
+//    else {
```

Comment:
Lets keep this comment undeleted. This includes a TODO comment we many need to revisit in future.

Reply:
Ok.